doc/design/python/caffe2_survey.md

```diff
@@ -0,0 +1,169 @@
+# Caffe2 Python实现调研
+
+## Caffe2 Python端主要模块
+![Python Concepts](http://api.paddlepaddle.org/graphviz?dot=https://gist.githubusercontent.com/reyoung/d9a9112730c514e0ee74aa5df8a8fa33/raw/fca60683d9b00eeb89f28e7065bd792b3b4ef4b4/caffe2_python.dot)
+
+Caffe2在Python端构建拓扑结构的概念主要如上图所示:
+
+1. Net: 一个神经网络拓扑
+2. ModelHelper: 包含神经网络拓扑，和他的参数以及参数初始化网络
+3. brew/optimizer/etc: 一些操作Model的帮助函数，也就是layer函数
+
+用户可以直接调用这三层逻辑的任何一层。并且，对于复杂的神经网络，用户**必须**直接调用最底层的Net或者ModelHelper才能完成。
+
+## 概念细节
+
+### Net
+
+[Net](https://github.com/caffe2/caffe2/blob/master/caffe2/python/core.py#L1298) 是Caffe2中最终生成神经网络拓扑结构的对象。使用Net进行训练的样例为(从官方[Toy Regression](https://github.com/caffe2/caffe2/blob/master/caffe2/python/tutorials/Toy_Regression.ipynb)中摘录，将ipynb转换成Python代码):
+
+```python
+from caffe2.python import core, workspace
+
+init_net = core.Net("init")
+
+# The ground truth parameters.
+W_gt = init_net.GivenTensorFill(
+    [], "W_gt", shape=[1, 2], values=[2.0, 1.5])
+B_gt = init_net.GivenTensorFill([], "B_gt", shape=[1], values=[0.5])
```

Comment:
because these Operators are used to init parameters, so they do not have input variables, only output variables, which is the second argument.